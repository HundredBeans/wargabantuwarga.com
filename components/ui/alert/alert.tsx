import * as React from "react";

import {
  accentBorderColors,
  alertColors,
  AlertProps,
  renderAlertIcon,
} from "./utils";

import clsx from "clsx";

export const Alert = React.forwardRef<HTMLDivElement, AlertProps>(
  (
    {
      children,
      className,
      color = "yellow",
      accentBorder = false,
      icon,
      ...rest
    },
    ref,
  ) => {
    const _className = clsx(
      alertColors(color),
      accentBorderColors(accentBorder, color),
      "rounded-md p-4",
      className,
    );

    return (
      <div className={_className} role="alert" {...rest} ref={ref}>
        <div className="flex">
          {icon && (
            <div className="flex-shrink-0 mr-3">{renderAlertIcon(icon)}</div>
          )}
          <div>{children}</div>
        </div>
      </div>
    );
  },
);

Alert.displayName = "Alert";
