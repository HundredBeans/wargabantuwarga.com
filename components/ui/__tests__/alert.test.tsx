import React from "react";

import { Alert } from "../alert";

import { render } from "@testing-library/react";

describe("Alert", () => {
  it("renders correctly", () => {
    const { container } = render(<Alert>Test Alert</Alert>);

    expect(container.firstChild).toMatchInlineSnapshot(`
  <div>
    <div
      class="bg-yellow-50 text-yellow-700 p-4"
      role="alert"
    >
      <div
        class="flex justify-between"
      >
        <div
          class="flex"
        >
          <div>
            Test Alert
          </div>
        </div>
      </div>
    </div>
  </div>
    `);
  });

  it("renders with correct customization", () => {
    const { container } = render(
      <Alert accentBorder className="babayaga" color="red" visible={true}>
        Test Customize Alert
      </Alert>,
    );

    expect(container.firstChild?.firstChild).toHaveClass(
      "bg-red-50",
      "text-red-700",
    );
    expect(container.firstChild?.firstChild).toHaveClass(
      "border-l-4",
      "border-red-400",
    );
    expect(container.firstChild?.firstChild).toHaveClass("p-4", "babayaga");
  });
});
