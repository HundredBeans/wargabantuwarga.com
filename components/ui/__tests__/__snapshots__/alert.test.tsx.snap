// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert customize 1`] = `
<div>
  <div
    class="bg-red-50 text-red-700 border-l-4 border-red-400 rounded-md p-4 newclass"
    role="alert"
  >
    <div
      class="flex justify-between"
    >
      <div
        class="flex"
      >
        <div>
          Test
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`loads and displays Alert component 1`] = `
<div>
  <div
    class="bg-yellow-50 text-yellow-700 rounded-md p-4"
    role="alert"
  >
    <div
      class="flex justify-between"
    >
      <div
        class="flex"
      >
        <div>
          Test Alert
        </div>
      </div>
    </div>
  </div>
</div>
`;
